common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,23 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
-    } else {
-      return text;
+    text = text.replace("''", "'");
+    if (isQuoted(text, "\"")) {
+      return text.substring(1, text.length() - 1).replace("\\\"", "\"").replace("\\\\", "\\");
+    }
+    if (isQuoted(text, "'")) {
+      return text.substring(1, text.length() - 1).replace("\\'", "'").replace("\\\\", "\\");
     }
+    if (isQuoted(text, "`")) {
+      return text.substring(1, text.length() - 1).replace("\\`", "`").replace("\\\\", "\\");
+    }
+    return text;
```

Comment:
Is it possible to reach this line?  
Maybe we need to throw an error instead (RuntimeError?)

Reply:
It'll reach it in the case where `unquoteText` is called when text doesn't need to be unquoted (or using quotes not supported, in which case it won't "unquote it"). I think it's a needed case.

Reply:
So we're calling unquote string on a string that doesn't need unquoting...?  That sounds wrong ;)

Reply:
It is being called on every thing. String that needed to be unquoted would be unquoted, the rest should left unchanged.